src/pages/api/webhooks/webhook.ts

```diff
@@ -1,6 +1,9 @@
 import type { NextApiRequest, NextApiResponse } from "next";
 import * as crypto from "crypto";
 import { env } from "~/env.mjs";
+import { sendEmail } from "~/utils/sendEmail";
+
+const receivedDeliveryIds = new Set();
```

Comment:
hm, this is in-memory

since you mention vercel is "serverless", does this state persist across multiple requests? if we have a new instance of the "server" for each request then this set will always be empty.

Reply:
Yes, this in in-memory and since vercel is serverless, the state does not persist across multiple requests when it is hosted. To solve this if the project is hosted, saving the delivery ID's in a database or using cloud storage would persist all the ID's and would be an effective solution.

Reply:
A CRON job can be used to delete the ID's periodically.